### PR TITLE
SmartBox message text color

### PR DESCRIPTION
**Pull request by ChrisAntaki**
Adds `messageTextColor` parameter to SmartBox's options

### CODE REVIEW

File: docs/buttons.md [hunk 1]
```diff
@@ -103,12 +102,10 @@ subscriptions.attachSmartButton(
...
-- `options` (Optional): Accepts JSON object with `theme` attribute.\
-                        i.e.: `{theme: 'dark'|'light'}` (Default is `light`) \
-                        `dark` theme renders the button with dark background and\
-                        light/white text.\
-                        `light` theme renders the button with light/white background \
-                        and dark text.
+- `options` (Optional): Configures appearance of button.
+  - `lang`: (Optional) Sets the button SVG and title.  English is default language.  See [Button API](../src/runtime/button-api.js) for `lang` values.
+  - `theme`: (Optional) Button theme can be `light` (default) or `dark`.
+  - `messageTextColor`: (Optional) Overrides theme color for message text. (ex: `'#09f'`)
```
Comment (dparikh) [inline docs/buttons.md]: We might want to add a sample image at the end with `transparent` and `messageTextColor` set, once the boq implementation is done.

**Comment by dparikh** [comment]
Before we submit, need to test along side boq changes.

File: examples/sample-pub/publisher.js [hunk 2]
```diff
@@ -118,7 +118,12 @@ function startFlowAuto() {
 
         subs.attachSmartButton(
             smartButton,
-            {theme: 'light', lang: 'en'},
+            {
+              theme: 'light',
+              lang: 'en',
+              backgroundColor: 'transparent',
```
[the remark this answers is not in the document]
Comment (ChrisAntaki) [reply examples/sample-pub/publisher.js]: Good catch! It's now removed

File: docs/buttons.md (hunk 1 above)
Comment (ChrisAntaki) [reply docs/buttons.md]: Sounds good!

File: src/runtime/smart-button-api.js [hunk 3]
```diff
@@ -74,13 +74,19 @@ export class SmartSubscriptionButtonApi {
...
+    const frontendArguments = {
       'productId': this.deps_.pageConfig().getProductId(),
       'publicationId': this.deps_.pageConfig().getPublicationId(),
       'theme': this.options_ && this.options_.theme || 'light',
       'lang': this.options_ && this.options_.lang || 'en',
-    });
+    };
+    const messageTextColor = this.options_ && this.options_.messageTextColor;
+    if (messageTextColor) {
+      frontendArguments['messageTextColor'] = messageTextColor;
```
Comment (jpettitt) [inline src/runtime/smart-button-api.js]: Do we need to validate this?

**Comment by ChrisAntaki** [comment]
Thanks for the review!

**Pull request merged**